e2e-tests/valset_update_test.go

```diff
@@ -124,3 +125,71 @@ func (suite *ConsumerKeeperTestSuite) TestUnbondMaturePackets() {
 	suite.Require().Equal(uint64(2), commitments[1].Sequence, "did not send VSCMatured packet for VSC packet 2")
 }
+
+// TestRedelegationNoConsumer tests a redelegate transaction submitted on a provider chain with no consumers
+func (s *ProviderTestSuite) TestRedelegationNoConsumer() {
+	providerKeeper := s.providerChain.App.(*appProvider.App).ProviderKeeper
+	stakingKeeper := s.providerChain.App.(*appProvider.App).StakingKeeper
+
+	// remove the consumer chain, which was already registered during setup
+	providerKeeper.DeleteConsumerClientId(s.providerCtx(), s.consumerChain.ChainID)
+
+	// Setup delegator and bond amount
+	bondAmt := sdk.NewInt(10000000)
+	delAddr := s.providerChain.SenderAccount.GetAddress()
+
+	// Delegate to validator 0
+	_, val0Addr := s.getVal(0)
+	val0TokensBefore := stakingKeeper.Validator(s.providerCtx(), val0Addr).GetBondedTokens()
+	val0PowerBefore := stakingKeeper.GetLastValidatorPower(s.providerCtx(), val0Addr)
+	_, shares, _ := delegate(s, delAddr, bondAmt)
+
+	// Assert expected amount was bonded to validator 0
+	val0TokensAfter := stakingKeeper.Validator(s.providerCtx(), val0Addr).GetBondedTokens()
+	s.Require().Equal(val0TokensAfter.Sub(val0TokensBefore), bondAmt)
+
+	s.providerChain.NextBlock()
+
+	// Check validator power
+	val0PowerAfter := stakingKeeper.GetLastValidatorPower(s.providerCtx(), val0Addr)
+	s.Require().Greater(val0PowerAfter, val0PowerBefore)
+
+	// Pick new validator for redelegation
+	_, val1Addr := s.getVal(1)
+	val1TokensBefore := stakingKeeper.Validator(s.providerCtx(), val1Addr).GetBondedTokens()
+	val1PowerBefore := stakingKeeper.GetLastValidatorPower(s.providerCtx(), val1Addr)
+
+	// redelegate shares to new validator
+	redelegate(s, delAddr,
+		val0Addr, // Source
+		val1Addr, // Dest
+		shares)
+
+	// Assert expected amount was delegated to validator 1
+	val1TokensAfter := stakingKeeper.Validator(s.providerCtx(), val1Addr).GetBondedTokens()
+	s.Require().Equal(val1TokensAfter.Sub(val1TokensBefore), bondAmt)
+
+	// Assert delegated tokens amount returned to original value for validator 0
+	s.Require().Equal(val0TokensBefore, stakingKeeper.Validator(s.providerCtx(), val0Addr).GetBondedTokens())
+
+	s.providerChain.NextBlock()
+
+	// 1 redelegation record should exist with appropriate maturation time
+	redelegations := stakingKeeper.GetRedelegations(s.providerCtx(), delAddr, 2)
+	s.Require().Len(redelegations, 1)
+	redelgation := redelegations[0]
+	providerUnbondingPeriod := stakingKeeper.UnbondingTime(s.providerCtx())
+	s.Require().Equal(
+		s.providerCtx().BlockTime().Add(providerUnbondingPeriod),
+		redelgation.Entries[0].CompletionTime)
+
+	// Check validator powers
+	val1PowerAfter := stakingKeeper.GetLastValidatorPower(s.providerCtx(), val1Addr)
+	s.Require().Greater(val1PowerAfter, val1PowerBefore)
+	s.Require().Equal(val0PowerBefore, stakingKeeper.GetLastValidatorPower(s.providerCtx(), val0Addr))
+}
```

Comment:
`TestRedelegationNoConsumer` should also check that after the unbonding period passes on the provider, the redelegation entry is no longer there. Use `TestUnbondingNoConsumer` as an example.